### PR TITLE
consistently use absolute paths

### PR DESCRIPTION
**Pull request by evmar**
Two commits, can be reviewed separately.

This is part of my yak shave for running tsickle tests in google3, but it also came up in an unrelated yak shave, so I may as well submit it.  If the approach looks good to you I'll need to test it against g3 and verify I didn't break any assumptions there as well.

### CODE REVIEW

File: src/externs.ts [hunk 1]
```diff
@@ -124,14 +124,9 @@ export function getGeneratedExterns(
...
- *
- * @param contextFileName if given is used as the context file path to resolve the source file's
- *     name against.
  */
-export function moduleNameAsIdentifier(
-    host: AnnotatorHost, fileName: string, contextFileName?: string): string {
-  const resolved = resolveModuleName(host, contextFileName || '', fileName);
-  return host.pathToModuleName('', resolved).replace(/\./g, '$');
+export function moduleNameAsIdentifier(host: AnnotatorHost, fileName: string): string {
+  return host.pathToModuleName('', fileName).replace(/\./g, '$');
```
Comment (mprobst) [inline src/externs.ts]: reminder: we'll need to make sure the bazel rules implementation (in the host) has the same assertion here before we land this.

File: test/BUILD [hunk 2]
```diff
@@ -30,6 +30,8 @@ ts_library(
 jasmine_node_test(
     name = "unit_test",
     srcs = [":unit_test_lib"],
+    # Unit tests need access to tslib.
```
Comment (mprobst) [inline test/BUILD]: shouldn't that rather be a "deps" entry on unit_test_lib then? data is not how TS dependencies are supposed to work.

File: test/e2e_main_test.ts [hunk 3]
```diff
@@ -24,25 +26,22 @@ describe('toClosureJS', () => {
     const result = toClosureJS(
         testSupport.compilerOptions, filePaths, {isTyped: true},
         (filePath: string, contents: string) => {
-          files.set(filePath, contents);
+          files.set(path.relative(testSupport.compilerOptions.rootDir!, filePath), contents);
```
Comment (mprobst) [inline test/e2e_main_test.ts]: Add a comment that this post-processes to be relative again?

File: test/test_support.ts [hunk 4]
```diff
@@ -19,8 +19,25 @@ import * as ts from 'typescript';
...
+ * the source tree set up under the hermetic testing environment.
+ */
+function rootDir(): string {
+  const runfiles = process.env['RUNFILES'];
+  assert(runfiles);
+  return path.join(runfiles!, 'tsickle');
+}
+
+/**
+ * Path to tslib.d.ts, relative to the project root.
```
Comment (mprobst) [inline test/test_support.ts]: it's absolute, isn't it?

File: test/test_support.ts [hunk 5]
```diff
@@ -19,8 +19,25 @@ import * as ts from 'typescript';
...
+ *
+ * The test suite runs the TS compiler, which wants to load tslib.
```
Comment (mprobst) [inline test/test_support.ts]: I don't understand why we need tslib to begin with - do have test code that has an import of tslib? that does seem odd.

File: test/test_support.ts [hunk 6]
```diff
@@ -37,22 +54,20 @@ export const baseCompilerOptions: ts.CompilerOptions = {
   importHelpers: true,
   noEmitHelpers: true,
   baseUrl: '.',
-  paths: tslibPath ? {
+  paths: {
     // The compiler builtin 'tslib' library is looked up by name,
     // so this entry controls which code is used for tslib.
     'tslib': [tslibPath]
-  } :
-                     undefined,
```
Comment (mprobst) [inline test/test_support.ts]: yay for fixing formatting by API change! :-)

File: test/BUILD (hunk 2 above)
Comment (evmar) [reply test/BUILD]: The tests run the TS compiler, which needs tslib.d.ts as an input data file.  I clarified the comment.

The way this works is that if you have importHelpers on, and your code does e.g.
  x = {...y}
then TS wants to use its tslib shim for Object.assign().  To do so it attempts to module resolve 'tslib'.  If tslib isn't found (or if it doesn't have a __assign export) then it's a compile error.

So for the tests to be able properly run TypeScript compilation, they need to ensure that the compiler they run succeeds in resolving tslib.

File: src/externs.ts (hunk 1 above)
Comment (evmar) [reply src/externs.ts]: Which assertion?  It appears this code works in google3 without modifications to its pathToModuleName.

**Comment by evmar** [comment]
Made one final change: the new arg to getGeneratedExterns is now optional, because Angular doesn't pass it and I can't simultaneously change Angular.

**Comment by evmar** [comment]
This appears to work in google3.

File: src/externs.ts (hunk 1 above)
Comment (mprobst) [reply src/externs.ts]: Yeah it works, but I think we should have the "all paths are absolute" assert in the g3 `pathToModuleName` as well, for safety - right?

Comment (evmar) [reply src/externs.ts]: Ah I see what you mean, yes.  Right now I still cannot assert paths are absolute:
https://github.com/angular/tsickle/pull/901/files#diff-1310126e2af8f00760656adfc34dc400R38

Our test suite actually manages to sneak the string "goog:foo.bar" into this function, which ends up getting canonicalized into a module name like 'goog_foo.bar' (!).  I want to clean that up to but I figured this change was a good stopping point that marked the next steps clearly.

**Pull request merged**